client/client_impl.go

```
@@ -76,9 +78,22 @@ func NewClientTokenTLS(uri, token string, c *tls.Config) Client {
 	auther := config.Client(oauth2.NoContext, &oauth2.Token{AccessToken: token})
 	if c != nil {
 		if trans, ok := auther.Transport.(*oauth2.Transport); ok {
-			trans.Base = &http.Transport{
-				TLSClientConfig: c,
-				Proxy:           http.ProxyFromEnvironment,
+			if os.Getenv("SOCKS_PROXY") != "" {
+				dialer, err := proxy.SOCKS5("tcp", os.Getenv("SOCKS_PROXY"), nil, proxy.Direct)
+				if err != nil {
+					fmt.Fprintln(os.Stderr, "can't connect to the proxy:", err)
+					os.Exit(1)
```

This should return the error. Calling `os.Exit` kills the entire program.